### PR TITLE
ref: upgrade django-stubs

### PR DESCRIPTION
**Pull request by asottile-sentry**

### CODE REVIEW

File: src/sentry/api/decorators.py [hunk 1]
```diff
@@ -19,11 +19,7 @@ def is_considered_sudo(request: Request) -> bool:
         or is_api_key_auth(request.auth)
         or is_api_token_auth(request.auth)
         or is_org_auth_token_auth(request.auth)
-        or (
-            request.user.is_authenticated
-            and not isinstance(request.user, AnonymousUser)
-            and not request.user.has_usable_password()
-        )
+        or (request.user.is_authenticated and not request.user.has_usable_password())
```
Comment (asottile-sentry) [inline src/sentry/api/decorators.py:22]: `is_authenticated` can now be used to descriminate the `request.user` union -- so mypy is mad about the impossibility of `AnonymousUser` after that check

File: src/sentry/sentry_apps/api/endpoints/sentry_app_publish_request.py [hunk 2]
```diff
@@ -58,8 +58,8 @@ def post(self, request: Request, sentry_app) -> Response:
                 status=400,
             )
 
-        assert isinstance(request.user, User) or isinstance(
-            request.user, RpcUser
+        assert isinstance(
+            request.user, (User, RpcUser)
```
Comment (asottile-sentry) [inline src/sentry/sentry_apps/api/endpoints/sentry_app_publish_request.py:62]: similarly, mypy now knows this fact statically (but is not smart enough to see the `isinstance(..., (several, types))`!)

File: src/sentry/sentry_apps/api/endpoints/sentry_apps_stats.py [hunk 3]
```diff
@@ -23,7 +23,7 @@ class SentryAppsStatsEndpoint(SentryAppsBaseEndpoint):
     def get(self, request: Request) -> Response:
         sentry_apps = (
             SentryApp.objects.filter(installations__date_deleted=None)
-            .annotate(Count("installations"))
+            .annotate(installations__count=Count("installations"))
```
Comment (asottile-sentry) [inline src/sentry/sentry_apps/api/endpoints/sentry_apps_stats.py:26]: django-stubs improved `annotate` support -- but doesn't support non-named arguments.  setting the name to the default name is the simplest fix

File: src/sentry/api/decorators.py (hunk 1 above)
Comment (markstory) [reply src/sentry/api/decorators.py:22]: > is_authenticated can now be used to descriminate the request.user union

This will be very useful :clap:

**Review by markstory — APPROVED** [review]

**Pull request merged**